ops.h

```diff
@@ -57,6 +57,16 @@ HWY_INLINE constexpr size_t MaxCols() {
   return 2048;
 }
 
+template <typename To, typename From>
+HWY_INLINE constexpr std::enable_if_t<
+    std::is_arithmetic_v<To> && std::is_arithmetic_v<From>, To>
+StaticCast(From from) noexcept {
+  if constexpr (std::is_unsigned_v<From> && std::is_floating_point_v<To>)
+    return static_cast<To>(static_cast<int64_t>(from));
```

Comment:
int64_t -> float is also somewhat expensive. How about we replace int64_t with `hwy::SignedFromSize<sizeof(from)>`? This evaluates to int for From=float.

Reply:
Ah, it seems I've created traits similar to hwy::SignedFromSize. No need to reinvent the wheel I'll just use that.
I think it would be `hwy::SignedFromSize<2 * sizeof(from)>` , right ?

Reply:
Just out of curiosity, I've experimented with this function on Compiler Explorer and noticed that on 64-bit machines, int64 can sometimes yield faster performance. What's your take on this?

```
float func1(size_t size)
{
    return static_cast<float>(static_cast<int64_t>(size));
}

float func2(size_t size)
{
    return static_cast<float>(static_cast<int>(size));
}
```

In X86-64:
```
**clang-14**
func1(unsigned long):                              # @func1(unsigned long)
        push    rbp
        mov     rbp, rsp
        mov     qword ptr [rbp - 8], rdi
        cvtsi2ss        xmm0, qword ptr [rbp - 8]
        pop     rbp
        ret
func2(unsigned long):                              # @func2(unsigned long)
        push    rbp
        mov     rbp, rsp
        mov     qword ptr [rbp - 8], rdi
        mov     rax, qword ptr [rbp - 8]
        cvtsi2ss        xmm0, eax
        pop     rbp
        ret
```
```
**clang-17**
func1(unsigned long):                              # @func1(unsigned long)
        push    rbp
        mov     rbp, rsp
        mov     qword ptr [rbp - 8], rdi
        cvtsi2ss        xmm0, qword ptr [rbp - 8]
        pop     rbp
        ret
func2(unsigned long):                              # @func2(unsigned long)
        push    rbp
        mov     rbp, rsp
        mov     qword ptr [rbp - 8], rdi
        mov     rax, qword ptr [rbp - 8]
        cvtsi2ss        xmm0, eax
        pop     rbp
        ret
```
```
**gcc-13**
func1(unsigned long):
        push    rbp
        mov     rbp, rsp
        mov     QWORD PTR [rbp-8], rdi
        mov     rax, QWORD PTR [rbp-8]
        pxor    xmm0, xmm0
        cvtsi2ss        xmm0, rax
        pop     rbp
        ret
func2(unsigned long):
        push    rbp
        mov     rbp, rsp
        mov     QWORD PTR [rbp-8], rdi
        mov     rax, QWORD PTR [rbp-8]
        pxor    xmm0, xmm0
        cvtsi2ss        xmm0, eax
        pop     rbp
        ret
```

In Arm:

```
**clang-17**
func1(unsigned long):                              // @func1(unsigned long)
        sub     sp, sp, #16
        str     x0, [sp, #8]
        ldr     d0, [sp, #8]
        fmov    x8, d0
        scvtf   s0, x8
        add     sp, sp, #16
        ret
func2(unsigned long):                              // @func2(unsigned long)
        sub     sp, sp, #16
        str     x0, [sp, #8]
        ldr     x8, [sp, #8]
        scvtf   s0, w8
        add     sp, sp, #16
        ret
```
```
**gcc-13**
func1(unsigned long):
        sub     sp, sp, #16
        str     x0, [sp, 8]
        ldr     x0, [sp, 8]
        scvtf   s0, x0
        add     sp, sp, 16
        ret
func2(unsigned long):
        sub     sp, sp, #16
        str     x0, [sp, 8]
        ldr     x0, [sp, 8]
        fmov    s0, w0
        scvtf   s0, s0
        add     sp, sp, 16
        ret
```

Reply:
Interesting, thanks for showing the exact codegen. It's not 100% obvious which is better because 64-bit cvtsi2ss has lower throughput, and we can afford an extra reg. Looks like same-size is indeed better on Arm, though, and it's not super time-critical, so let's go with your current code 👍